presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1651,6 +1656,11 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
                         .collect(Collectors.toList())));
     }
 
+    public static boolean createFilesForMissingBuckets(Table table, ConnectorSession session)
+    {
+        return !(table.getTableType().equals(TEMPORARY_TABLE) && !createEmptyBucketFilesForTemporaryTable(session));
```

Comment:
nit: It is probably just a matter of taste, but it feels like `!table.getTableType().equals(TEMPORARY_TABLE) || createEmptyBucketFilesForTemporaryTable(session)` will be more readable (avoid double negation)

Reply:
Updated